In [1]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [2]:
%cd 'drive/My Drive'

/content/drive/My Drive


In [3]:
import pandas as pd
import numpy as np


In [4]:
#Loading the Data
df=pd.read_csv('AirplaneAccident.csv')

In [5]:
df.head()

Severity  ...  Accident_ID
0                Minor_Damage_And_Injuries  ...         7570
1                Minor_Damage_And_Injuries  ...        12128
2        Significant_Damage_And_Fatalities  ...         2181
3  Significant_Damage_And_Serious_Injuries  ...         5946
4        Significant_Damage_And_Fatalities  ...         9054

[5 rows x 12 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Severity                 10000 non-null  object 
 1   Safety_Score             10000 non-null  float64
 2   Days_Since_Inspection    10000 non-null  int64  
 3   Total_Safety_Complaints  10000 non-null  int64  
 4   Control_Metric           10000 non-null  float64
 5   Turbulence_In_gforces    10000 non-null  float64
 6   Cabin_Temperature        10000 non-null  float64
 7   Accident_Type_Code       10000 non-null  int64  
 8   Max_Elevation            10000 non-null  float64
 9   Violations               10000 non-null  int64  
 10  Adverse_Weather_Metric   10000 non-null  float64
 11  Accident_ID              10000 non-null  int64  
dtypes: float64(6), int64(5), object(1)
memory usage: 937.6+ KB


We have No Missing Values


In [7]:
df.describe()

Safety_Score  ...   Accident_ID
count  10000.000000  ...  10000.000000
mean      41.876406  ...   6266.554200
std       16.138072  ...   3610.170288
min        0.000000  ...      2.000000
25%       30.593607  ...   3139.750000
50%       41.278539  ...   6280.500000
75%       52.511416  ...   9391.500000
max      100.000000  ...  12500.000000

[8 rows x 11 columns]

In [8]:
df.corr()

Safety_Score  ...  Accident_ID
Safety_Score                 1.000000  ...    -0.001334
Days_Since_Inspection       -0.685386  ...     0.000582
Total_Safety_Complaints      0.057726  ...     0.004527
Control_Metric               0.000564  ...     0.006481
Turbulence_In_gforces        0.019603  ...    -0.014396
Cabin_Temperature            0.032747  ...     0.014865
Accident_Type_Code           0.173930  ...    -0.012278
Max_Elevation                0.004451  ...     0.008576
Violations                   0.041735  ...    -0.002047
Adverse_Weather_Metric      -0.107925  ...     0.008586
Accident_ID                 -0.001334  ...     1.000000

[11 rows x 11 columns]

In [11]:
df.isna().sum()

Severity                   0
Safety_Score               0
Days_Since_Inspection      0
Total_Safety_Complaints    0
Control_Metric             0
Turbulence_In_gforces      0
Cabin_Temperature          0
Accident_Type_Code         0
Max_Elevation              0
Violations                 0
Adverse_Weather_Metric     0
Accident_ID                0
dtype: int64

In [12]:
#Encoding the target values
from sklearn.preprocessing import LabelEncoder
labelencoder=LabelEncoder()
df['Severity']=labelencoder.fit_transform(df['Severity'])

In [29]:
#Encoded Labels
labelencoder.classes_

array(['Highly_Fatal_And_Damaging', 'Minor_Damage_And_Injuries',
       'Significant_Damage_And_Fatalities',
       'Significant_Damage_And_Serious_Injuries'], dtype=object)

In [13]:
y=df['Severity']

In [14]:
x=df.drop(['Severity'],axis=1)

In [15]:
from sklearn.model_selection import train_test_split
import random
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=random.randint(0,100))

In [16]:
print("Size of x_train: ",len(x_train),"Size of y_train: ",len(y_train))

Size of x_train:  8000 Size of y_train:  8000


In [17]:
print("Size of x_test: ",len(x_test),"Size of y_test: ",len(y_test))

Size of x_test:  2000 Size of y_test:  2000


For building the Model,I will be using the LightGradientBoostingMachine Algorithm

In [18]:
#Building the Model
import lightgbm as lgb

In [19]:
#Dataset for the LightGradientBoostingMachine
d_train=lgb.Dataset(x_train,label=y_train)

In [20]:
#Setting the parameters for the model
params = {}
params['num_boost_round']=2000
params['stratified']=True
params['nfold']=5

params['learning_rate'] = 0.05
params['boosting_type'] = 'gbdt'
params['objective'] = 'multiclass'
params['num_class']=4
params['metric'] = 'multi_error'

params['num_leaves'] = 60
params['max_depth'] = 9
params['feature_fraction']=0.45
params['bagging_fraction']=0.3

params['reg_alpha']=0.15
params['reg_lambda']=0.15
params['min_child_weight']=0

In [21]:
clf = lgb.train(params, d_train, 100)

/usr/local/lib/python3.6/dist-packages/lightgbm/engine.py:118: UserWarning: Found `num_boost_round` in params. Will use it instead of argument
  warnings.warn("Found `{}` in params. Will use it instead of argument".format(alias))


In [22]:
y_pred=clf.predict(x_test)

In [23]:
y_pred_index=np.argmax(y_pred,1)

In [24]:
y_test=y_test.to_numpy()

In [25]:
#Printing the Confusion Matrix
from sklearn.metrics import confusion_matrix,classification_report,accuracy_score
confusion_matrix(y_test,y_pred_index)

array([[549,  17,   8,   4],
       [  7, 488,   6,   7],
       [ 20,   1, 319,   4],
       [  8,  18,  13, 531]])

In [30]:
#Printing The Classification Report
target_name=['{}'.format(i) for i in labelencoder.classes_]
print(classification_report(y_test,y_pred_index,target_names=target_name))


                                         precision    recall  f1-score   support

              Highly_Fatal_And_Damaging       0.94      0.95      0.94       578
              Minor_Damage_And_Injuries       0.93      0.96      0.95       508
      Significant_Damage_And_Fatalities       0.92      0.93      0.92       344
Significant_Damage_And_Serious_Injuries       0.97      0.93      0.95       570

                               accuracy                           0.94      2000
                              macro avg       0.94      0.94      0.94      2000
                           weighted avg       0.94      0.94      0.94      2000



In [27]:
#Printing the Accuracy Score
accuracy_score(y_test,y_pred_index)

0.9435

We got an Accuracy of around 94% on the Test Set